main.go

```diff
@@ -1,65 +1,83 @@
 package main
 
 import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
 	"net/http"
-	"os"
-	"time"
+	"errors"
+	"github.com/gin-gonic/gin"
+	"io/ioutil"
+	"encoding/json"
 )
 
 type WeatherData struct {
 	City string `json:"city"`
-	Info string `json:"info"`
+	Country string `json:"country"`
+	Temp_C float64 `json:"temp_c"`
+	Temp_F float64 `json:"temp_f"`
+	Wind_KPH float64 `json:"wind_kph"`
+	Wind_MPH float64 `json:"wind_mph"`
 }
 
-var apiKey = os.Getenv("bc94089fef9d46968ff8d724889f9fff")
+func getWeatherByCity(city string) (*WeatherData, error) {
+	url := "https://weatherapi-com.p.rapidapi.com/current.json?q=" + city
 
-func main() {
-	http.HandleFunc("/weather/", func(w http.ResponseWriter, r *http.Request) {
-		city := r.URL.Path[len("/weather/"):]
+	req, _ := http.NewRequest("GET", url, nil)
 
-		url := "https://weatherapi-com.p.rapidapi.com/current.json?q=" + city
+	req.Header.Add("X-RapidAPI-Key", "356fcc1b96msh6b0261faa77fcd2p101d17jsnbbb68ac024b7")
+	req.Header.Add("X-RapidAPI-Host", "weatherapi-com.p.rapidapi.com")
 
-		req, _ := http.NewRequest("GET", url, nil)
+	res, err := http.DefaultClient.Do(req)
 
-		req.Header.Add("X-RapidAPI-Key", "356fcc1b96msh6b0261faa77fcd2p101d17jsnbbb68ac024b7")
-		req.Header.Add("X-RapidAPI-Host", "weatherapi-com.p.rapidapi.com")
+	if err != nil {
+		return nil, errors.New("Error getting weather data")
+	}
 
-		res, err := http.DefaultClient.Do(req)
+	defer res.Body.Close()
+	body, _ := ioutil.ReadAll(res.Body)
 
-		if err != nil {
-			fmt.Println("HTTP call failed:", err)
-			return
-		}
+	var data map[string]interface{}
+    err2 := json.Unmarshal([]byte(string(body)), &data)
 
-		defer res.Body.Close()
-		body, _ := ioutil.ReadAll(res.Body)
+    if err2 != nil {
+        return nil, errors.New("Error getting weather data")
+    }
 
-		weatherData := WeatherData{
-			City: city,
-			Info: string(body),
-		}
-		jsonData, _ := json.Marshal(weatherData)
-		ioutil.WriteFile("weather.json", jsonData, 0600)
+	var weatherData WeatherData
 
-		data, err := ioutil.ReadFile("weather.json")
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
+	if location, ok := data["location"].(map[string]interface{}); ok {
+		if current, ok := data["current"].(map[string]interface{}); ok {
+			weatherData := WeatherData{
+				City: location["name"].(string),
+				Country: location["country"].(string),
+				Temp_C: current["temp_c"].(float64),
+				Temp_F: current["temp_f"].(float64),
+				Wind_KPH: current["wind_kph"].(float64),
+				Wind_MPH: current["wind_mph"].(float64),
+			}
+			return &weatherData, nil
+		} else {
+			return nil, errors.New("Error getting weather data")
 		}
+	} else {
+		return nil, errors.New("Error getting weather data")
+	}
 
-		fmt.Fprintf(w, string(data))
-	})
+	return &weatherData, nil

```

Comment:
You are handling parsing errors, but not API errors. E.g., if you get rate limited you could repeat the query after a while, and it doesn't make sense to accept new ones during this time.